src/type.c

```diff
@@ -30,13 +32,21 @@ SEXP vec_ptype(SEXP x, struct vctrs_arg* x_arg) {
   case vctrs_type_character:   return vec_ptype_slice(x, vctrs_shared_empty_chr);
   case vctrs_type_raw:         return vec_ptype_slice(x, vctrs_shared_empty_raw);
   case vctrs_type_list:        return vec_ptype_slice(x, vctrs_shared_empty_list);
-  case vctrs_type_dataframe:   return bare_df_map(x, &col_ptype);
+  case vctrs_type_dataframe:   return df_ptype(x, true);
   case vctrs_type_s3:          return s3_type(x, x_arg);
   case vctrs_type_scalar:      stop_scalar_type(x, x_arg);
   }
   never_reached("vec_ptype");
 }
 
+// [[ include("vctrs.h") ]]
+SEXP vec_ptype_unnamed(SEXP x, struct vctrs_arg* x_arg) {
+  SEXP out = PROTECT(vec_ptype(x, x_arg));
+  out = vec_set_names(out, R_NilValue);
+  UNPROTECT(1);
+  return out;
+}
+
 static SEXP col_ptype(SEXP x) {
   return vec_ptype(x, args_empty);
```

Comment:
Should the columns of the data frame be named? Or should names be stripped from them too?

Reply:
Good point. I think they should be unnamed for consistency with vec-assign.